Azure/sonic-utilities
# Platform Driver Development Framework (PDDF): Adding PDDF CLI utils

**- What I did**
This change is related to Platform Driver Development Framework (PDDF) which is being added to sonic-buildimage repo. More details can be found here,
https://github.com/Azure/SONiC/pull/406


PDDF supports its own CLI utilities, which use generic PDDF component plugins. I added these PDDF CLI utilities.
 
**- How I did it**
Created pddf_psuutil, pddf_fanutil, pddf_thermalutil, pddf_ledutil.

**- How to verify it**
Under non-PDDF mode, these utilities won't run,
```
root@sonic:/home/admin# pddf_psuutil mfrinfo
PDDF mode should be supported and enabled for this platform for this operation
root@sonic:/home/admin#
root@sonic:/home/admin# pddf_fanutil status
PDDF mode should be supported and enabled for this platform for this operation
root@sonic:/home/admin#
```

In PDDF mode, these utilities can be run with root privileges.
 
**- Previous command output (if the output of a command-line utility has changed)**

**- New command output (if the output of a command-line utility has changed)**
```
root@sonic:/home/admin# pddf_fanutil getspeed
FAN      Front Fan RPM    Rear Fan RPM
-----  ---------------  --------------
FAN 1             8700            7400
FAN 2             8600            7300
FAN 3             8800            7500
FAN 4             8800            7500
FAN 5             8500            7300
FAN 6             8600            7300
root@sonic:/home/admin#
root@sonic:/home/admin#
root@sonic:/home/admin# pddf_fanutil status
FAN    Status
-----  --------
FAN 1  OK
FAN 2  OK
FAN 3  OK
FAN 4  OK
FAN 5  OK
FAN 6  OK
root@sonic:/home/admin# pddf_psuutil mfrinfo
PSU 1 is OK
Manufacture Id: 3Y POWER
Model: YM-2651
Serial Number: SA290N091739133083
Fan Direction: INTAKE

PSU 2 is Not OK

root@sonic:/home/admin#
```
-->
